test-framework/junit5/src/main/java/io/quarkus/test/junit/AbstractQuarkusTestWithContextExtension.java

```diff
@@ -47,6 +49,19 @@ protected void setState(ExtensionContext context, QuarkusTestExtensionState stat
         getStoreFromContext(context).put(QuarkusTestExtensionState.class.getName(), state);
     }
 
+    protected void clearState(ExtensionContext context) {
+        QuarkusTestExtensionState state = getState(context);
+        if (state != null) {
+            try {
+                state.close();
```

Comment:
I don't think this should be done in `afterAll` of a test class

Reply:
This is being executed only in the test main class (not in nested classes). 
Yet why do you think we should not be doing so?

Reply:
What happens when there are multiple tests being run? Under normal circumstances the context would not be cleared, but now with this change it will be.

Reply:
I think the issue may be related to mixing @QuarkusTest and @QuarkusIntegrationTest

I would try to sort tests, execute those annotated with @QuarkusTest then clear the context and execute integration tests

Reply:
The issue is that the state is wrongly shared among QuarkusTest and QuarkusIntegrationTest, the order does not really matter here.
With these changes, the state is never shared but for some reason, it makes tests to fail. I'm thinking to reject these changes and open a new pull request to have a single state per Quarkus test type, this should also fix the issue.